stackstate_checks_base/tests/test_http_helper.py

```diff
@@ -0,0 +1,1375 @@
+import unittest
+import requests_mock
+import json
+import six
+from requests import Session, Request
+from schematics.models import Model
+from schematics.types import StringType, IntType, BooleanType
+from stackstate_checks.utils.http_helper import HTTPHelper, HTTPRequestType, HTTPAuthenticationType, HTTPResponseType
+from stackstate_checks.utils.http_helper import HTTPHelperRequestHandler, HTTPHelperSessionHandler, HTTPMethod
+from stackstate_checks.utils.http_helper import HTTPHelperResponseHandler, HTTPHelperConnectionHandler, HTTPHelperCommon
+from requests.auth import HTTPBasicAuth
+from requests.packages.urllib3.util.retry import Retry
+
+
+"""
+Structures used within the test cases
+"""
+
+
+class BodySchematicTest(Model):
+    hello = IntType(required=True)
+
+
+class BodyRequestSchematicTest(Model):
+    hello = StringType(required=True)
+    pong = BooleanType(required=True)
+
+
+class BodyResponseSchematicTest(Model):
+    hello = StringType(required=True)
+    pong = BooleanType(required=True)
+
+
+"""
+ HTTP Helper Common Class
+"""
+
+
+class TestHTTPHelperCommon(unittest.TestCase):
```

Comment:
please also write a test with some weird unicode characters in both the request and the response.

Reply:
Will do